userApi/src/test/java/cs4337/group9/userapi/Service/UserServiceTest.java

```diff
@@ -0,0 +1,198 @@
+        });
+        assertEquals("User already exists with email: test@example.com", exception.getMessage());
+    }
+
+    @Test
+    void testCreateUser_Success() {
+        // Arrange
+        when(userRepository.findByEmail(user.getEmail())).thenReturn(Optional.empty());
+        when(userRepository.save(any(UserEntity.class))).thenReturn(user);
+
+        // Act
+        userService.createUser(user);
+
+        // Assert
+        verify(userRepository, times(1)).save(user);
+    }
+
+    @Test
+    void testGetUserById_UserNotFound_ThrowsException() {
+        // Arrange
+        when(userRepository.findById(userId)).thenReturn(Optional.empty());
+
+        // Act & Assert
+        UserNotFoundException exception = assertThrows(UserNotFoundException.class, () -> {
+            userService.getUserById(userId);
+        });
+        assertEquals("User not found with id: " + userId, exception.getMessage());
+    }
+
+    @Test
+    void testGetUserById_Success() {
+        // Arrange
+        when(userRepository.findById(userId)).thenReturn(Optional.of(user));
+
+        // Act
+        UserEntity result = userService.getUserById(userId);
+
+        // Assert
+        assertNotNull(result);
+        assertEquals(user.getId(), result.getId());
+    }
+
+    @Test
+    void testUpdateUser_UserNotFound_ThrowsException() {
+        // Arrange
+        UserEntity updatedUser = new UserEntity(userId, "updatedUser", "updated@example.com", "newpassword", Role.USER, Status.ACTIVE, "Updated about", "updated.jpg", null, null);
+        when(userRepository.findById(userId)).thenReturn(Optional.empty());
+
+        // Act & Assert
+        UserNotFoundException exception = assertThrows(UserNotFoundException.class, () -> {
+            userService.updateUser(userId, updatedUser);
+        });
+        assertEquals("User not found with id: " + userId, exception.getMessage());
+    }
+
+    @Test
+    void testUpdateUser_Success() {
+        // Arrange
+        UserEntity updatedUser = new UserEntity(userId, "updatedUser", "updated@example.com", "newpassword", Role.USER, Status.ACTIVE, "Updated about", "updated.jpg", null, null);
+        when(userRepository.findById(userId)).thenReturn(Optional.of(user));
+        when(userRepository.save(any(UserEntity.class))).thenReturn(updatedUser);
+
+        // Act
+        userService.updateUser(userId, updatedUser);
+
+        // Assert
+        verify(userRepository, times(1)).save(updatedUser);
+        assertEquals(updatedUser.getUsername(), user.getUsername());
+    }
+
+    @Test
+    void testDeleteUser_UserNotFound_ThrowsException() {
+        // Arrange
+        when(userRepository.existsById(userId)).thenReturn(false);
+
+        // Act & Assert
+        UserNotFoundException exception = assertThrows(UserNotFoundException.class, () -> {
+            userService.deleteUser(userId);
+        });
+        assertEquals("User not found with id: " + userId, exception.getMessage());
```

Comment:
The `assertEquals()` message doesnt match the message acutally thrown by the exception 
`Expected :User not found with id: 9347bba9-0e8a-4225-842b-0811413b7d2d
Actual   :User with ID 9347bba9-0e8a-4225-842b-0811413b7d2d not found`